src/lib/network_pool/transaction_pool.ml

```diff
@@ -1165,157 +1167,153 @@ struct
+                                                 when adding from gossip"
+                                                ~metadata:
+                                                  [ ( "error"
+                                                    , `String
+                                                        (Verifier
+                                                         .invalid_to_string
+                                                           invalid) )
+                                                  ] ;
+                                              add_failure
+                                                (Invalid_failure invalid) ;
+                                              None
+                                          | Ok (Ok [ c ]) ->
+                                              Some c
+                                          | Ok (Ok _) ->
+                                              assert false)
+                                        (`Unchecked
+                                          ( Transaction_hash.User_command.create
+                                              uc
+                                          , c ))
+                                        account.nonce
+                                        (Currency.Balance.to_amount
+                                           (balance_of_account ~global_slot
+                                              account))
                                     with
-                                    | `Reject ->
-                                        add_failure (Command_failure e) ;
-                                        Mutex.release signer_lock ;
-                                        return (Error `Invalid_command)
-                                    | `Ignore ->
-                                        go sender_local_state u_acc acc
-                                          ( ( uc
-                                            , diff_error_of_indexed_pool_error e
-                                            )
-                                          :: rejected )
-                                          cs )
-                                | Ok (res, sender_local_state, u) ->
-                                    let%bind _ =
+                                    | Error e -> (
+                                        match%bind
+                                          handle_command_error t ~trust_record
+                                            ~is_sender_local uc e
+                                        with
+                                        | `Reject ->
+                                            add_failure (Command_failure e) ;
+                                            return (Error `Invalid_command)
+                                        | `Ignore ->
+                                            go sender_local_state u_acc acc
+                                              ( ( uc
```

Comment:
In the `apply` function should we release locks for accounts from the list of txns that are ignored but don't cause the entire diff to be rejected? We only look at the accepted list there